### PR TITLE
Add own implementation for public suffix list

### LINKED ISSUE #232: DNS stats should be based upon public suffix
DNS stats are calculated in buckets of `Top QName 2` and `Top QName 3`, but this does not work with second-level domain names like .co.uk. It would be better if these stats are calculated based upon the public suffix list (so queries to www.google.co.uk would count towards stats as www.google.co.uk and google.co.uk and not google.co.uk and .co.uk).

Please see https://publicsuffix.org/list/

### PR DESCRIPTION
**Pull request by leoparente**

### CODE REVIEW

**Review by rboucher-me — APPROVED** [review]
Did not validate the code itself, but the implementation looks good to me.

File: src/handlers/dns/DnsStreamHandler.cpp [hunk 1]
```diff
@@ -335,14 +334,18 @@ bool DnsStreamHandler::_filtering(DnsLayer &payload, [[maybe_unused]] PacketDire
...
+            if (ends_with(qname_ci, fqn)) {
                 _static_suffix_size = fqn.size();
                 goto will_not_filter;
             }
         }
         // checked the whole list and none of them matched: filter
         goto will_filter;
     }
+    // This is not actually a filter, but only should work if OnlyQNameSuffix is not enabled
+    else if (_f_enabled[Filters::PublicSuffixList] && payload.parseResources(true) && payload.getFirstQuery() != nullptr) {
```
Comment (weyrick) [inline src/handlers/dns/DnsStreamHandler.cpp]: did you use _f_enabled for convenience? i would rather not conflate it with config options which are not filters. if it makes sense we can add _c_enabled and a similar list of config functionality

File: src/handlers/dns/DnsStreamHandler.cpp [hunk 2]
```diff
@@ -335,14 +334,18 @@ bool DnsStreamHandler::_filtering(DnsLayer &payload, [[maybe_unused]] PacketDire
...
+        _static_suffix_size = match_public_suffix(payload.getFirstQuery()->getNameLower());
```
Comment (weyrick) [inline src/handlers/dns/DnsStreamHandler.cpp]: remind me - we cache this lower name right?

**Review by weyrick — CHANGES_REQUESTED** [review]

File: src/handlers/dns/DnsStreamHandler.cpp (hunk 2 above)
Comment (leoparente) [reply src/handlers/dns/DnsStreamHandler.cpp]: yes, we do

File: src/handlers/dns/DnsStreamHandler.cpp (hunk 1 above)
Comment (leoparente) [reply src/handlers/dns/DnsStreamHandler.cpp]: yes. i did for convenience. I will change that

Comment (leoparente) [reply src/handlers/dns/DnsStreamHandler.cpp]: done

**Review by weyrick — APPROVED** [review]

**Pull request merged**